src/openms/include/OpenMS/QC/MzCalibration.h

```diff
@@ -0,0 +1,83 @@
+//    without specific prior written permission.
+// For a full list of authors, refer to the file AUTHORS.
+// --------------------------------------------------------------------------
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL ANY OF THE AUTHORS OR THE CONTRIBUTING
+// INSTITUTIONS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+// OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+// ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// --------------------------------------------------------------------------
+// $Maintainer: Chris Bielow $
+// $Authors: Juliane Schmachtenberg $
+// --------------------------------------------------------------------------
+
+#pragma once
+#include <OpenMS/QC/QCBase.h>
+
+namespace OpenMS
+{
+  class FeatureMap;
+  class MSExperiment;
+
+  /**
+    @brief QC metric calculating uncalibrated m/z error
+
+    Metric sets m/z-values of the original experiment and the calculated reference m/z-values, uncalibrated m/z error (ppm)
+    and calibrated m/z error (ppm) as meta-values of the PeptideIdentification in post FDR FeatureMaps.
+
+    **/
+  class OPENMS_DLLAPI MzCalibration : public QCBase
+  {
+    public:
+      /// Constructor
+      MzCalibration();
+
+      /// Destructor
+      virtual ~MzCalibration() = default;
+
+      /**
+        @brief find original m/z Value, set meta values "mz_raw", "mz_ref", "uncalibrated_mz_error_ppm", "calibrated_mz_error_ppm"
+        @param exp Peak map of the original experiment with original m/z-value before calibration generated by InternalCalibration
+        @param features  contains m/z-value of PeptideIdentification after calibration
+       **/
+      void compute(FeatureMap& features, const MSExperiment& exp);
+
+      /// define the required input filed MzML before Calibration, FeatureXML after FDR
+      Status requires() const override;
+
+      /// Returns the name of the metric.
+      const String& getName() const override;
+
+    private:
+      /// search matching RT-time in MSExperiment before calibration, and return the m/z value. Search with error tolerance EPSILON
+      double getMZraw_(double rt, const MSExperiment& exp) const;
+
+      double mz_raw_;
+      double mz_ref_;
+      bool no_mzml_;
+      String name_;
+  };
+  /// EPSILON: error tolerance for RT-searching in MSExperiment
+  static const double EPSILON_{ 0.05 };
```

Comment:
defining this in a header globally is a no-go.
You can define it in the cpp though.
Also remove the mention of `EPSILON` from `getMZraw_()` and just say ` a small tolerance window of ...`